packages/react/src/checkbox/Checkbox.tsx

```diff
@@ -0,0 +1,132 @@
+import * as RadixCheckbox from "@radix-ui/react-checkbox";
+import * as RadixLabel from "@radix-ui/react-label";
+import { useId } from "@reach/auto-id";
+import { type ComponentPropsWithRef, type ReactNode, forwardRef } from "react";
+
+import type { ExtendProps } from "../utils";
+
+import { Box } from "../box";
+import { Flex } from "../flex";
+import { Text } from "../text";
+import * as styles from "./Checkbox.css";
+
+type CheckboxProps = ExtendProps<
+  ComponentPropsWithRef<typeof Box>,
+  {
+    checked?: "indeterminate" | boolean;
+    children: ReactNode;
+    defaultChecked?: "indeterminate" | boolean;
+    disabled?: boolean;
+    readonly?: boolean;
+  }
+>;
+
+export const Checkbox = forwardRef<HTMLDivElement, CheckboxProps>(
+  (
+    {
+      checked,
+      children,
+      className,
+      defaultChecked,
+      disabled,
+      id: idProp,
+      readonly = false,
+      ...props
+    },
+    ref,
+  ) => {
+    const id = useId(idProp);
+
+    return (
+      <Flex
+        data-disabled={disabled || readonly}
+        ref={ref}
+        {...props}
+        {...styles.wrapper({}, className)}
```

Comment:
the order here needs to be reversed so we allow consumers to override our own props
```suggestion
        {...styles.wrapper({}, className)}
        {...props}
```

Reply:
fixed bhai